docs/data/data-grid/column-menu/column-menu.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Data Grid - Column menu
+---
+
+# Data grid - Column menu
+
+<p class="description">Customize your columns menu.</p>
+
+## Column menu
+
+Each column header comes with a column menu with quickly accessible grid features like column visibility, sorting, filtering, and others.
+
+It can be accessed by clicking on 3-dots icon that appear on hover on a header cell.
+
+{{"demo": "ColumnMenuGrid.js", "bg": "inline"}}
+
+## Disable column menu
+
+By default, each column header has the column menu enabled. To disable the column menu, set the prop `disableColumnMenu={true}`.
+
+{{"demo": "DisabledColumnMenuGrid.js", "bg": "inline"}}
+
+## Customize column menu
+
+You can customize the column menu either by using `components`, `componentsProps` and `initialItems` supported by column menu component or by using custom component using the `ColumnMenu` slot of the data grid component.
+
+### Add, override, hide and reorder items
+
+Column menu component accept `components`, `componentsProps` and `initialItems` using which you can:
+
+- Add new items
+- Override default items
+- Hide default items
+- Reorder items
+
+**components**: Prop accepted by column menu components `<GridColumnMenuDefault />` or `<GridColumnMenuSimple />`, could be used to override default column menu components or add new components.
+
+Default components supported by community package are `ColumnMenuSortItem`, `ColumnMenuFilterItem`, `ColumnMenuHideItem` and `ColumnMenuColumnsItem`. Pro package adds `ColumnMenuPinningItem` and Premium package adds `ColumnMenuAggregationItem` and `ColumnMenuGroupingItem`.
```

Comment:
We can remove the list of default components here, since there's a table containing it in a more structured form